pysmt/formula.py

```diff
@@ -519,6 +519,24 @@ def Max(self, *args):
             h = len(exprs) // 2
             return self.Max(self.Max(exprs[0:h]), self.Max(exprs[h:]))
 
+    def EqualsOrIff(self, left, right):
+        """Returns Equals() or Iff() depending on the type of the arguments.
+
+        This can be used to deal with ambiguous cases where we might be
+        dealing with both Theory and Boolean atoms.
+        """
+        if left.is_symbol():
+            type_ = left.symbol_type()
+        elif right.is_symbol():
```

Comment:
You can do the same with constants, but all of this should be unnecessary, as all types are cached